modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -357,6 +382,12 @@
         this.logTxRecords = IgniteSystemProperties.getBoolean(IGNITE_WAL_LOG_TX_RECORDS, false);
 
         cctx.txMetrics().onTxManagerStarted();
+
+        keyCollisionsInfo = new KeyCollisionsHolder();
```

Comment:
I do not see where else this link is written, I think it should be done final without need for volatile read.

Reply:
disagree, reader thread can be differ.

Reply:
This field of IgniteTxManager and I don't see a problem making it final.

Reply:
Wrong, here we need cctx, which we get in start method.